packages/url_launcher/url_launcher/android/src/main/java/dev/flutter/plugins/urllauncher/UrlLauncherPlugin.java

```diff
@@ -0,0 +1,58 @@
+package dev.flutter.plugins.urllauncher;
+
+import android.support.annotation.NonNull;
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.embedding.engine.plugins.activity.ActivityAware;
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
+
+/**
+ * Plugin implementation that uses the new {@code io.flutter.embedding} package.
+ *
+ * <p>Instantiate this in an add to app scenario to gracefully handle activity and context changes.
+ */
+public final class UrlLauncherPlugin implements FlutterPlugin, ActivityAware {
```

Comment:
Does this class need a static registerWith() method?

Reply:
I put it in a `UrlLauncherPluginRegistrar` class. I'm not 100% on that being the best place for it, but I wanted to move it as far away from here as possible to try and make it harder to accidentally initialize the "old" vs the "new" version of this.

Reply:
Never mind, our plugins infra expects Android and iOS to use the same name for the class that holds `registerWith`. I moved it back.